uportal-war/src/main/java/org/jasig/portal/rest/PortletsRESTController.java

```diff
@@ -104,17 +107,60 @@ public ModelAndView getManageablePortlets(HttpServletRequest request, HttpServle
      */
     @RequestMapping(value="/portlet/{fname}.json", method = RequestMethod.GET)
     public ModelAndView getPortlet(HttpServletRequest request, HttpServletResponse response, @PathVariable String fname) throws Exception {
-      IPerson user = personManager.getPerson(request);
-      EntityIdentifier ei = user.getEntityIdentifier();
-      IAuthorizationPrincipal ap = AuthorizationService.instance().newPrincipal(ei.getKey(), ei.getType());
-      IPortletDefinition portletDef = portletDefinitionRegistry.getPortletDefinitionByFname(fname);
-      if(portletDef != null && ap.canRender(portletDef.getPortletDefinitionId().getStringId())) {
-        LayoutPortlet portlet = new LayoutPortlet(portletDef);
-        return new ModelAndView("json", "portlet", portlet);
-      } else {
-        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
-        return new ModelAndView("json");
-      }
+        IAuthorizationPrincipal ap = getAuthorizationPrincipal(request);
+        IPortletDefinition portletDef = portletDefinitionRegistry.getPortletDefinitionByFname(fname);
+        if(portletDef != null && ap.canRender(portletDef.getPortletDefinitionId().getStringId())) {
+            LayoutPortlet portlet = new LayoutPortlet(portletDef);
+            return new ModelAndView("json", "portlet", portlet);
+        } else {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return new ModelAndView("json");
+        }
+    }
+
+    /**
+     * Provides a single, fully-rendered portlet.  NOTE:  Access to this API
+     * enpoint requires only <code>IPermission.PORTAL_SUBSCRIBE</code> permission.
+     */
+    @RequestMapping(value="/v4-3/portlet/{fname}.html", method = RequestMethod.GET)
```

Comment:
Since this is the most recent version, thoughts on utilzing multiple URLS?

`@RequestMapping(value={"/v4-3/portlet/{fname}.html", "/portlet/{fname}.html"})` for example.  Then when we get another version we can just switch the base one to that one along with v4-10 or something.

Also not sure i like the dash in the url, but that is just an opinion.